x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -227,31 +229,52 @@ public void getConnectorSyncJob(String connectorSyncJobId, ActionListener<Connec
      * @param listener               The action listener to invoke on response/failure.
      */
     public void cancelConnectorSyncJob(String connectorSyncJobId, ActionListener<UpdateResponse> listener) {
-        Instant cancellationRequestedAt = Instant.now();
+        try {
+            getConnectorSyncJob(connectorSyncJobId, listener.delegateFailure((getSyncJobListener, syncJobSearchResult) -> {
+                String timestampFieldToUpdate;
+                Instant updatedTimestamp = Instant.now();
+
+                ConnectorSyncStatus nextStatus;
+                ConnectorSyncStatus prevStatus = getConnectorSyncJobStatusFromSearchResult(syncJobSearchResult);
+
+                if (ConnectorSyncStatus.PENDING.equals(prevStatus)) {
```

Comment:
Any reason we just have this logic for `PENDING` and not `SUSPENDED`? I think we should also cancel suspended sync like that to replicate Kibana logic: https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/lib/cancel_syncs.ts#L32

Reply:
Kibana seems to take the responsibility of putting any pending and suspended jobs into a canceled stated for a user-initiated sync canceling.

Reply:
Thanks for pointing this out! Then I'll update the logic to also consider `SUSPENDED` and to also update `completed_at` and `cancelation_requested_at`.

Reply:
Adjusted with [702e2f8](https://github.com/elastic/elasticsearch/pull/105722/commits/702e2f8d0f04e21ddb194f7a68cb0c1ec030bee9)